support/netutil/netutil.go

```diff
@@ -72,3 +72,64 @@ func watchSignalWithinTimeout(timeout time.Duration, errSignal chan struct{}, ac
 	}
 	action()
 }
+
+// get the local area network IP address
+func GetLocalHost() string {
+	addresses, err := net.InterfaceAddrs()
+	if err != nil {
+		fmt.Printf("net.InterfaceAddrs failed, err: %v", err.Error())
+		return ""
+	}
+
+	for _, address := range addresses {
+		if ipNet, ok := address.(*net.IPNet); ok && !ipNet.IP.IsLoopback() {
+			// IPv4
+			if ipNet.IP.To4() != nil {
+				return ipNet.IP.String()
+			}
+			// IPv6
+			if ipNet.IP.To16() != nil && ipNet.IP.To4() == nil {
+				return ipNet.IP.String()
+			}
+		}
+	}
+	fmt.Printf("no network interface found")
+	return ""
+}
+
+// provide default values for host and port
+func GetHostAndIP(bindStr string, portStr string) (host string, port int) {
+	host = "localhost"
+	localIp := GetLocalHost()
+	validHosts := map[string]bool{
+		"localhost": true,
+		"0.0.0.0":   true,
+		localIp:     true,
+		"127.0.0.1": true,
```

Comment:
"::1" also no?

You are only considering IPv4 binding

Reply:
I didn't consider ipv6, I will try to fix it. Thank you.